File: benchmarks/spec/concurrency/simple_api_call.rb
require 'ruby-jmeter'

test do

  defaults  domain: 's3-ap-southeast-2.amazonaws.com',
            protocol: 'https'

  with_user_agent :iphone

  header [
    { name: 'Accept-Encoding', value: 'gzip,deflate,sdch' },
    { name: 'Accept', value: 'text/javascript, text/html, application/xml, text/xml, */*' }
  ]

  threads count: 5000,
          rampup: 240,
          scheduler: true,
          duration: 1800,
          continue_forever: true,
          delayedStart: true  do

    constant_timer delay: 10000

    get name: 'api', url: '/flood-loadtest/api.json'
  end
end.jmx
# end.flood(ENV['FLOOD_API_TOKEN'], {
#   name: 'High API Concurrency JMeter',
#   tag_list: 'benchmarks',
#   privacy_flag: 'public'})

File: benchmarks/spec/concurrency/simple_api_call.scala
import io.gatling.core.Predef._
import io.gatling.http.Predef._
import scala.concurrent.duration._

import flood._

class TestPlan extends Simulation {

  val httpConf = httpConfigFlood
    .baseURL("https://s3-ap-southeast-2.amazonaws.com/")
    .disableCaching
    .acceptHeader("text/javascript, text/html, application/xml, text/xml, */*;")
    .acceptEncodingHeader("gzip, deflate")

  val scn = scenario("benchmark")
    .during(1800 seconds) {
      exec(http("api")
        .get("/flood-loadtest/api.json"))
      .pause(10000 milliseconds)
    }

  setUp(scn.inject(rampUsers(5000) over (240 seconds))).protocols(httpConf)
}
